src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs

```diff
@@ -46,11 +46,34 @@ public static ExecutionPlanNode ConvertShowPlanTreeToExecutionPlanTree(Node curr
                 Properties = GetProperties(currentNode.Properties),
                 Children = currentNode.Children.Select(x => ConvertShowPlanTreeToExecutionPlanTree(x)).ToList(),
                 Edges = currentNode.Edges.Select(x => ConvertShowPlanEdgeToExecutionPlanEdge(x)).ToList(),
+                OverlayIcons = GenerateNodeOverlay(currentNode),
                 Name = currentNode.DisplayName,
                 ElapsedTimeInMs = currentNode.ElapsedTimeInMs
             };
         }
 
+        public static List<ExecutionPlanNodeOverlay> GenerateNodeOverlay(Node currentNode)
```

Comment:
You can make this an extension method for nodes.

Reply:
I do not want to mix contract classes with showplan classes.

Reply:
In what way would they mix? The extension method would go in a separate static class of its own and would only be for the `Node` type. You wouldn't have to change the `Node` class itself.

Reply:
I am a little confused by the suggestion. 

1. If the suggestion is to add it to ShowPlan/Node class

The badge feature seems more like an ExecutionPlan specific functionality and not a showplan functionality. In the past, I have tried to keep the showplan code as close to ssms as possible and just add my own logic to execution plan graph utils. It follows a similar pattern to that. Also, I do not see any additional benefit of adding badges method to the node class. 

2. If the suggestion is to add a setter method to ExecutionPlanNode class 
The parameters of the setter might be confusing then. What will be the input. Because providing a showplan/node as an input to the setter does not seem  intuitive

Reply:
He's just saying make this the method (notice the "this" added)

`public static List<Badge> GenerateNodeOverlay(this Node node)`

and then you call it like this 

`node.GenerateNodeOverlay()` - although if you're doing it like this I would actually rename it to 

`public static List<Badge> GetBadges(this Node node)` 

It doesn't really change any of the logic. Extension methods are just nicer to use and read for situations like this.

(if you do this as he suggested normally extension methods go in their own separate static classes)